tortoise/threshold.go

```diff
@@ -31,32 +32,35 @@ func computeBallotWeight(
 				targetWeight = atxweight
 			}
 		}
-
 		expected, err := proposals.GetNumEligibleSlots(targetWeight, total, layerSize, layersPerEpoch)
 		if err != nil {
 			return weight{}, fmt.Errorf("unable to compute number of eligibile ballots for atx %s", ballot.AtxID)
 		}
-		rst := weightFromUint64(targetWeight)
-		rst = rst.div(weightFromUint64(uint64(expected)))
-		weights[ballot.ID()] = rst
-		return rst, nil
-	}
-	if ballot.RefBallot == types.EmptyBallotID {
-		return weight{}, fmt.Errorf("empty ref ballot and no epoch data on ballot %s", ballot.ID())
-	}
-	rst, exist := weights[ballot.RefBallot]
-	if !exist {
-		refballot, err := bdp.GetBallot(ballot.RefBallot)
-		if err != nil {
-			return weight{}, fmt.Errorf("ref ballot %s for %s is unknown", ballot.ID(), ballot.RefBallot)
+		reference = weightFromUint64(targetWeight)
+		reference = reference.div(weightFromUint64(uint64(expected)))
```

Comment:
don't understand the need for `referenceWeights`. 

this PR makes a miner produce one ballot per layer. this means this ballot takes up all the weight for this layer.
there is no need to call `GetNumEligibleSlots` anymore. can assign `weights[ballot.ID()] = targetWeight` directly.

Reply:
but weight per layer is not constant. it depends on the number of eligibilities (number of ballots)

Reply:
before you did `targetWeight/numEligibleSlots` for that ballot. now that this ballot has `numEligibleSlots` combined in one, why isn't it `targetWeight` now for that ballot? or are you saying the previous code is incorrect?

Reply:
targetWeight is for the epoch. numEligibleSlots is for the epoch. by dividing them up we get weight for a single eligibility (single ballot). previously we had eligibility = ballot. therefore we could store it in the same map. now every ballot has a variable number of eligibilities, so same map doesn't work. at least this was my underestanding.

Reply:
my confusion came from mixing up numSlots is for the whole epoch, not just a layer. what you said made sense.